hathor/utils/weight.py

```diff
@@ -0,0 +1,23 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from math import exp, log
+
+_LN_2 = log(2)
+
+
+def weight_to_work(weight: float) -> int:
+    """Convert weight to work rounding to the nearest integer."""
+    # return int(0.5 + 2**weight)
+    return int(0.5 + exp(weight * _LN_2))
```

Comment:
Which one has greater precision?

Reply:
I've decided to go with `math.ceil(2**weight)`, it's exactly the same as method1, but doesn't need the `0.5 +` operation.